main_f4.c

```diff
@@ -331,6 +416,19 @@ main(void)
 	/* do board-specific initialisation */
 	board_init();
 
+	/* configure the clock for bootloader activity */
+	rcc_clock_setup_hse_3v3(&clock_setup);
+
+	/* start the interface */
+#if INTERFACE_USART
+	cinit(BOARD_INTERFACE_CONFIG_USART, USART);
```

Comment:
Is the break detect unreliable? If it is just sampling the IO pin would help decomplicate reverting of the SoC to the reset state. If we do not have to init the uart, but just sample the pins. 

To further decouple things. We could use the SysTic timer to time the RX pin instead of timer 2. It is running at 1ms but can be sampled - using code from https://github.com/PX4/Firmware/blob/nuttx_next/src/drivers/bootloaders/include/timer.h#L341-L451 and https://github.com/PX4/Firmware/blob/nuttx_next/src/drivers/bootloaders/src/arch/stm32/drivers/can/driver.c#L154-L191

We can time the low time of the RX pin.